resources/lib/services/library_updater.py

```diff
@@ -44,69 +59,61 @@ def is_idle(self):
             self.startidle = lastidle
         if lastidle < self.startidle:
             self.startidle = 0
-        idletime = lastidle - self.startidle
-        return idletime >= 300
+        return (lastidle - self.startidle) >= 300
 
-    def library_update_scheduled(self):
+    def onSettingsChanged(self):
         """
-        Checks if the scheduled time for a library update has been reached
+        As settings changed, we will compute next schedule again
+        to ensure it's still correct
         """
-        try:
-            now = datetime.now()
-            update_frequency = g.ADDON.getSettingInt('auto_update')
-            interval = g.ADDON.getSettingInt('schedule_check_interval')
-            next_schedule_check = (self.last_schedule_check +
-                                   timedelta(minutes=interval))
-
-            if not update_frequency or now <= next_schedule_check:
-                return False
-
-            self.last_schedule_check = now
-            time = g.ADDON.getSetting('update_time') or '00:00'
-            lastrun_date = (g.ADDON.getSetting('last_update') or
-                            '1970-01-01')
-            lastrun = common.strp('{} {}'.format(lastrun_date, time[0:5]),
-                                  '%Y-%m-%d %H:%M')
-            nextrun = lastrun + timedelta(days=[0, 1, 2, 5, 7][update_frequency])
-            common.log(
-                'It\'s currently {}, next run is scheduled for {}'
-                .format(now, nextrun))
-
-            return now >= nextrun
-        except TypeError:
-            # When there is concurrency between getSettingX and setSettingX at the same time,
-            # the get settings fails to read
-            return False
-
-
-    def onScanStarted(self, library):
+        self.next_schedule = _compute_next_schedule()
+
+    def onScanStarted(self, lib):
```

Comment:
the call interface use 'library' not 'lib'
why you have changed?

Reply:
Because library masks the libraty global name (import resources.lib.kodi.library as library).
So to avoid confusion.

Reply:
so is more correct to use the right interface property names
instead the "import" can change name as you wish like

from: import resources.lib.kodi.library as library
to: import resources.lib.kodi.library as lib_library

this variable is used only once if I've seen it right